crypto/ec/ec_ameth.c

```diff
@@ -196,14 +196,14 @@ static int eckey_priv_encode(PKCS8_PRIV_KEY_INFO *p8, const EVP_PKEY *pkey)
     }
     p = ep;
     if (!i2d_ECPrivateKey(&ec_key, &p)) {
-        OPENSSL_free(ep);
+        OPENSSL_clear_free(ep, eplen);
```

Comment:
It would make sense to simplify the code here to avoid double call to i2d_ECPrivateKey(). If we avoid allocating ep and just set p to NULL before `i2d_ECPrivateKey()` the i2d function will allocate things for us. We can then just OPENSSL_clear_free() in the error case of `PKCS8_pkey_set0()`.

Reply:
> It would make sense to simplify the code here to avoid double call to i2d_ECPrivateKey(). If we avoid allocating ep and just set p to NULL before `i2d_ECPrivateKey()` the i2d function will allocate things for us. We can then just OPENSSL_clear_free() in the error case of `PKCS8_pkey_set0()`.

--Updated,Thanks